File: frontend/src/services/review.ts
/**
 * 评分评论服务
 */
import { api } from './api'

export interface Review {
  id: number
  user_id: number
  lesson_id: number
  rating: number
  comment: string | null
  created_at: string
  updated_at: string
}

export interface ReviewWithUser extends Review {
  user_name: string
  user_avatar: string | null
}

export interface ReviewCreate {
  lesson_id: number
  rating: number
  comment?: string
}

export interface ReviewUpdate {
  rating: number
  comment?: string
}

export interface LessonRatingStats {
  lesson_id: number
  average_rating: number
  review_count: number
  rating_distribution: Record<number, number>
}

class ReviewService {
  /**
   * 创建评论
   */
  async createReview(data: ReviewCreate): Promise<Review> {
    return await api.post<Review>('/reviews/', data)
  }

  /**
   * 更新评论
   */
  async updateReview(reviewId: number, data: ReviewUpdate): Promise<Review> {
    return await api.put<Review>(`/reviews/${reviewId}`, data)
  }

  /**
   * 删除评论
   */
  async deleteReview(reviewId: number): Promise<void> {
    await api.delete(`/reviews/${reviewId}`)
  }

  /**
   * 获取课程的所有评论
   */
  async getLessonReviews(lessonId: number): Promise<ReviewWithUser[]> {
    return await api.get<ReviewWithUser[]>(`/reviews/lesson/${lessonId}`)
  }

  /**
   * 获取课程的评分统计
   */
  async getLessonRatingStats(lessonId: number): Promise<LessonRatingStats> {
    return await api.get<LessonRatingStats>(`/reviews/lesson/${lessonId}/stats`)
  }

  /**
   * 获取我对某课程的评论
   */
  async getMyReview(lessonId: number): Promise<Review | null> {
    return await api.get<Review | null>(`/reviews/my/${lessonId}`)
  }
}

export const reviewService = new ReviewService()

File: backend/app/schemas/review.py
"""
评分评论相关的Pydantic Schema
"""

from datetime import datetime
from typing import Optional, Dict
from pydantic import BaseModel, Field, validator


class ReviewCreate(BaseModel):
    """创建评论"""

    lesson_id: int
    rating: int = Field(..., ge=1, le=5, description="评分1-5")
    comment: Optional[str] = None


class ReviewUpdate(BaseModel):
    """更新评论"""

    rating: int = Field(..., ge=1, le=5, description="评分1-5")
    comment: Optional[str] = None


class ReviewResponse(BaseModel):
    """评论响应"""

    id: int
    user_id: int
    lesson_id: int
    rating: int
    comment: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ReviewWithUser(ReviewResponse):
    """带用户信息的评论"""

    user_name: str
    user_avatar: Optional[str]


class LessonRatingStats(BaseModel):
    """课程评分统计"""

    lesson_id: int
    average_rating: float
    review_count: int
    rating_distribution: Dict[int, int]  # {1: 10, 2: 20, 3: 30, 4: 40, 5: 50}
